src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -102,6 +102,29 @@ inline __device__ __host__ int8_t cast_to(const int32_t& val)
     return static_cast<int8_t>(val & 0xff);
 }
 
+inline __device__ __host__ bool IsZero(double val) { return val == 0.0; }
+
+inline __device__ __host__ bool IsOne(double val) { return val == 1.0; }
+
+template <typename dst_data_t, typename acc_data_t>
+inline __device__ void applyalphaBetaUpdate(dst_data_t* __restrict__ p_array,
+                                            const acc_data_t value,
+                                            double alpha,
+                                            double beta,
+                                            size_t index)
+{
+    if(IsOne(alpha) && IsZero(beta))
+    {
+        p_array[index] = cast_to<acc_data_t, dst_data_t>(value);
+        return;
+    }
+
+    double val_alpha_beta =
+        cast_to<double, acc_data_t>(alpha) * value +
+        cast_to<double, acc_data_t>(p_array[index]) * cast_to<double, acc_data_t>(beta);
+    p_array[index] = cast_to<double, dst_data_t>(val_alpha_beta);
```

Comment:
It should be
```C++
    acc_data_t val_alpha_beta =
        cast_to<double, acc_data_t>(alpha) * value +
        cast_to<dst_data_t, acc_data_t>(p_array[index]) * cast_to<double, acc_data_t>(beta);
    p_array[index] = cast_to<acc_data_t, dst_data_t>(val_alpha_beta);
```
- for `val_alpha_beta`, `double` is excessive since all the computations are performed in `acc_data_t` domain
- the value read from `p_array[index]` is of the `dst_data_t` type, not `double`.

Reply:
[correctness] was fixed in latest develop branch https://github.com/ROCm/MIOpen/blob/develop/src/kernels/gpu_reference_kernel/naive_conv.cpp#L122C5-L124C66

Reply:
> [performance] you recommend float?

`acc_data_t val_alpha_beta = ...`